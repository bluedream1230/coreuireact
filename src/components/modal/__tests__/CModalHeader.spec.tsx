import * as React from 'react'
import { render /* ,screen */, fireEvent /* ,waitFor */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CModalHeader } from '../../../index'

test('loads and displays CModalHeader component', async () => {
  const { container } = render(<CModalHeader>Test</CModalHeader>)
  expect(container).toMatchSnapshot()
})

test('CModalHeader customize', async () => {
  const { container } = render(<CModalHeader className="bazinga">Test</CModalHeader>)
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('bazinga')
  expect(container.firstChild).toHaveClass('modal-header')
})

test('CModalHeader click on button close', async () => {
  const onDismiss = jest.fn()
  render(
    <CModalHeader className="bazinga" onDismiss={onDismiss}>
      Test
    </CModalHeader>,
  )
  expect(onDismiss).toHaveBeenCalledTimes(0)
  const btn = document.querySelector('.btn-close')
  if (btn !== null) {
    fireEvent.click(btn)
  }
  expect(onDismiss).toHaveBeenCalledTimes(1)
})
