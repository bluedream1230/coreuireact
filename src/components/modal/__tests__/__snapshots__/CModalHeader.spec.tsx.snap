// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CModalHeader customize 1`] = `
<div>
  <div
    class="modal-header bazinga"
  >
    Test
  </div>
</div>
`;

exports[`loads and displays CModalHeader component 1`] = `
<div>
  <div
    class="modal-header"
  >
    Test
  </div>
</div>
`;
